src/aws/flb_aws_credentials.c

```diff
@@ -546,6 +546,7 @@ int flb_read_file(const char *path, char **out_buf, size_t *out_size)
 
     fp = fopen(path, "r");
     if (!fp) {
+        fclose(fp);
```

Comment:
this is not valid. if ```fp``` is NULL, you cannot close it because is not referencing an open file.

Reply:
the error reported is on line 555